packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -1257,8 +1272,12 @@ subtask(TASK_COMPILE_SOLIDITY_LOG_COMPILATION_RESULT)
   .addParam("quiet", undefined, undefined, types.boolean)
   .setAction(
     async ({ compilationJobs }: { compilationJobs: CompilationJob[] }) => {
-      if (compilationJobs.length > 0) {
-        console.log("Solidity compilation finished successfully");
+      const count = compilationJobs.length;
+
+      if (count > 0) {
+        console.log(
+          `Compiled ${count} Solidity ${pluralize(count, "file")} successfully`
```

Comment:
This is printing the number of compilation jobs, not the number of compiled files.